strax/corrections.py

```diff
@@ -0,0 +1,120 @@
+'''I/O format for corrections using MongoDB
+
+This module list, reads, writes, among others functionalities
+to a MongoDB where corrections are stored.
+'''
+
+from datetime import datetime
+import logging
+import pandas as pd
+import pdmongo as pdm
+import pymongo
+
+import strax
+
+export, __all__ = strax.exporter()
+
+
+@export
+class CorrectionsInterface:
+    '''A class to manage corrections that are stored in a MongoDB.
+    '''
+
+    def __init__(self, host='127.0 0.1', username=None, password=None,
+                 database_name='corrections'):
+
+        self.host = host
+        self.username = username
+        self.password = password
+        self.client = pymongo.MongoClient(host=self.host,
+                                          username=self.username,
+                                          password=self.password)
+        self.database_name = database_name
+        self.database = self.client[self.database_name]
+
+    def list_corrections(self):
+        '''Smart logic to list corrections.
+        '''
+        return [x['name'] for x in self.database.list_collections()
+                if 'global' not in x['name']]
+
+    def read(self, correction):
+        '''Smart logic to read corrections,
+        where correction is the name of the DataFrame.
+        '''
+        df = pdm.read_mongo(correction, [], self.database)
+
+        # No data found
+        if df.size == 0:
+            return None
+        # Delete internal Mongo identifier
+        del df['_id']
+        return df.set_index('time')
+
+    def interpolate(self, what, when, how='interpolate'):
+        '''Interpolate values of a given correction(what=DataFrame)
+        given a time(when). For information of interpolation methods see,
+        https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.interpolate.html
+        '''
+        df = what
+
+        df_new = pd.DataFrame.from_dict({'Time': [when]})
+
+        df_new = df_new.set_index('Time')
+
+        df_combined = pd.concat([df, df_new], sort=False)
+
+        df_combined = df_combined.sort_index()
+        if how == 'interpolate':
+            df_combined = df_combined.interpolate(method='linear')
+        elif how == 'fill':
+            df_combined = df_combined.ffill()
+        else:
+            raise ValueError('Specify an interpolation method')
```

Comment:
Maybe be a bit more explicit here and tell the user which options he/she has.

Reply:
Done